### PR TITLE
Allow Tyger to run in Docker

### PR DESCRIPTION
**Pull request by johnstairs**
This change allows you to install Tyger in a local Docker environment instead of the cloud. When installed, you get a number of containers:

- tyger-local-db: a PostreSQL database
- tyger-data-plane: implements a subset of the Azure blob service for buffers
- tyger-control-plane: the Tyger server
- tyger-local-gateway: used on non-Linux environments to access Unix domain sockets.

We are also introducing "ephemeral" buffers, where the data is relayed to/from the run container without being persisted.

This is designed to be local to Docker host and uses standard Unix permissions on a domain sockets for security. To access this from a remote machine, you can do `tyger login ssh://myhost` and every command will use SSH to execute on the Tyger/SSH host.

Distributed runs and resource governance are not supported, but otherwise the Docker mostly implements the same functionality as the cloud version. We therefore run the same integration tests in multiple contexts (skipping them where the functionality is not implemented or not applicable). To develop in Docker mode, run `make open-docker-window` and the build and test makefile commands are implemented in the Docker context.

### CODE REVIEW

**Review by hansenms — COMMENTED** [review]
This is really a nice feature. There is a lot going on there, but the things I thought might be complicated (like the local dataplane provider) is not really, and things that I thought would be simple (like a run, which conceptually is really just a container), has a lot more moving parts, but it all makes sense. 

Some stuff for future work:
* I think we need a way to exclude GPUs that should not be used by the local system. Specifically if you have a workstation with a display GPU and a compute GPU, you want to just never assign the display one to a running container. 
* A nice thing to have would be a method for packaging images for import on an offline system. We need the tyger system images, which are know at build time, but also perhaps images for runs.....the ones in the codespecs. It could really just be a simple bash script but since the CLI knows about the system images, it could know how to do this. 
* The need for deleting buffers is much more apparent when running locally. So some kind of TTL on the buffers and an ability to purge would be really great.

File: README.md [hunk 1]
```diff
@@ -9,9 +9,10 @@ application is not limited to MRI and it could be used in a variety of domains
 and scenarios.
 
 At a high level, Tyger is a REST API that abstracts over an Azure Kubernetes
-cluster and Azure Blob storage. Future plans include support for on-prem
```
[the remark this answers is not in the document]
Comment (johnstairs) [reply README.md:12]: I think it all works now.

File: cli/cmd/buffer-sidecar/main.go [hunk 2]
```diff
@@ -33,26 +40,200 @@ func newRootCommand() *cobra.Command {
...
+						readerChan <- os.Stdin
+					}
+					log.Info().Msgf("Opening file %s for reading", inputFilePath)
+					inputFile, err := openFileFunc(inputFilePath, os.O_RDONLY, 0)
+					if err != nil {
+						if err == context.Canceled {
+							log.Warn().Msg("OpenFile operation canceled. Will return an empty response body.")
+							readerChan <- io.NopCloser(bytes.NewReader([]byte{}))
+							go func() {
+								time.Sleep(time.Minute)
```
[the remark this answers is not in the document]
Comment (johnstairs) [reply cli/cmd/buffer-sidecar/main.go:184]: Added a comment that hopefully clarifies

**Review by hansenms — APPROVED** [review]
Aside from he hang I described on Windows, this is good to go.

**Pull request merged**